src/backend/distributed/planner/relation_restriction_equivalence.c

```diff
@@ -2053,21 +2056,30 @@ ContextCoversJoinRestriction(JoinRestrictionContext *joinRestrictionContext,
 			continue;
 		}
 
+		List *joinRestrictInfoListInTest =
+			joinRestrictionInTest->joinRestrictInfoList;
+		bool joinIsOnTrue = list_length(joinRestrictInfoListInTest) == 0;
```

Comment:
I wonder if this is specific for "on true" or should we give it a bit more generic name

Reply:
perhaps just something like `hasJoinRestriction = list_length(joinRestrictInfoListInTest) > 0`